byway/src/main/webapp/recommendations.js

```diff
@@ -206,25 +217,32 @@ function placeMarker(place) {
   };
   let marker = new google.maps.Marker({
     position: place.geometry.location,
-    map,
+    map: canvas,
     title: place.name,
     icon: image
   });
   marker.addListener("click", () => {
     infoWindow.open(map, marker);
   });
+  marker.addListener("dblclick", () => {
+    route.push(place);
+    marker.setMap(null);
+    calcRouteWithRecs();
+  });
   markers.push(marker);
 }
 
 /**
- * Iterate through all markers and hide or show based on the boolean
- * pass in. Either set the marker canvas to null with false, or set
- * to the google.maps object with true.
- * @param {Boolean} isNull indicates to remove markers by setting their map to null.
+ * Iterate through all markers and hide or show based on the 
+ * boolean tracking the toggle state. Only affect rec markers,
+ * not ones added to the route.
  */
-function showMarkers(isNull) {
-  let canvas = (isNull) ? null : map;
+function toggleRecMarkers() {
+  let canvas = (isHiding) ? null : map;
```

Comment:
Nitt: would be useful to have a small comment on why this works. Why does setting the map to null hide the marker. Extra context in an inline comment would be nice.

Reply:
Fixed